src/class-utils.php

```diff
@@ -147,4 +147,15 @@ public static function get_edit_or_view_link( $post ) {
 		}
 		return $title;
 	}
+
+	/**
+	 * Gets the ID of the original post intended to be rewritten with the copy for Rewrite & Republish.
+	 *
+	 * @param int $post_id The copy post ID.
+	 *
+	 * @return int The original post id of a copy for Rewrite & Republish.
+	 */
+	public static function get_original_post_id( $post_id ) {
```

Comment:
Why not use `Utils::get_original()` and then get the `ID` from the resulting post object?

Reply:
`Utils::get_original()` returns  `WP_Post|null` so before getting the ID I'd need to check for `null` for all the three times it's used in this class. Seems to me that would be unnecessarily repeated code. Instead, `Utils::get_original_post_id()` already does what I need. It always returns an `int`. In fact, if `get_post_meta()` fails, `false` is casted to a zero integer .